gflags fails to build on fresh ubuntu14 box

gflags fails to build on vanilla ubuntu14 box on vagrant/virtualbox.

```
Compiling the CXX compiler identification source file "CMakeCXXCompilerId.cpp" failed.
Compiler: /usr/bin/g++
Build flags: -fPIE;-fPIC;-O2;-DNDEBUG;-march=x86-64;-mtune=i386;-mno-avx
Id flags:

The output was:
1
CMakeCXXCompilerId.cpp:1:0: error: CPU you selected does not support x86-64 instruction set
 /* This source file must have a .cpp extension so that all C++ compilers
 ^


Compiling the CXX compiler identification source file "CMakeCXXCompilerId.cpp" failed.
Compiler: /usr/bin/g++
Build flags: -fPIE;-fPIC;-O2;-DNDEBUG;-march=x86-64;-mtune=i386;-mno-avx
Id flags: -c

The output was:
1
CMakeCXXCompilerId.cpp:1:0: error: CPU you selected does not support x86-64 instruction set
```

I did upgrade Vagrant and VirtualBox to the latest releases. VirtualBox 5.0 does support avx instructions, but that shouldn't matter since we have `-mno-avx`

```bash
vagrant@vagrant-ubuntu-trusty-64:/vagrant$ cat /proc/cpuinfo | grep flags
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc pni pclmulqdq ssse3 cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx rdrand lahf_lm abm avx2
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc pni pclmulqdq ssse3 cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx rdrand lahf_lm abm avx2
```

It does build fine on the centos7.1 box. I did destroy the ubuntu14 box and started again with fresh box but not much help.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -149,7 +149,7 @@ else()
 endif()
 
 if(NOT DEFINED ENV{OPTIMIZED})
-  add_compile_options(-march=x86-64 -mtune=i386 -mno-avx)
+  add_compile_options(-march=x86-64 -mno-avx)
 endif()
 
 # make analyze (environment variable from Makefile)

--- a/tools/provision.sh
+++ b/tools/provision.sh
@@ -9,7 +9,7 @@
 
 set -e
 
-CFLAGS="-fPIE -fPIC -O2 -DNDEBUG -march=x86-64 -mtune=i386 -mno-avx"
+CFLAGS="-fPIE -fPIC -O2 -DNDEBUG -march=x86-64 -mno-avx"
 SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 BUILD_DIR="$SCRIPT_DIR/../build"
 WORKING_DIR="/tmp/osquery-provisioning"

